terminalcraft/app/gallery/page.tsx

```diff
@@ -0,0 +1,1222 @@
+    .replace(/\[([^\]]+)\]\([^)]+\)/g, '$1') // Remove links, keep text
+    .replace(/```[\s\S]*?```/g, '') // Remove code blocks
+    .replace(/^\s*[-*+]\s+/gm, '') // Remove list markers
+    .replace(/^\s*\d+\.\s+/gm, '') // Remove numbered list markers
+    .replace(/\n+/g, ' ') // Replace newlines with spaces
+    .trim();
+  
+  return plainText.length > maxLength 
+    ? plainText.slice(0, maxLength) + '...'
+    : plainText;
+}
+
+async function detectProjectLanguage(projectName: string): Promise<string> {
+  try {
+    const response = await fetch(
+      `https://api.github.com/repos/hackclub/terminalcraft/contents/submissions/${projectName}`,
+      {
+        headers: getGitHubHeaders()
+      }
+    );
+    
+    if (!response.ok) {
+      return 'Unknown';
+    }
+    
+    const files = await response.json();
+    const fileExtensions: { [key: string]: number } = {};
+    
+    // Count file extensions and special files
+    files.forEach((file: any) => {
+      if (file.type === 'file') {
+        const fileName = file.name.toLowerCase();
+        
+        // Handle files with extensions
+        if (fileName.includes('.')) {
+          const ext = fileName.split('.').pop();
+          if (ext) {
+            fileExtensions[ext] = (fileExtensions[ext] || 0) + 1;
+          }
+        } 
+        // Handle special files without extensions (common in CLI projects)
+        else {
+          // Common CLI executable/script names
+          const specialFiles: { [key: string]: string } = {
+            'makefile': 'makefile',
+            'dockerfile': 'docker',
+            'cargo': 'cargo',
+            'gemfile': 'gemfile',
+            'pipfile': 'pipfile',
+            'requirements': 'requirements',
+            'package': 'package',
+            'go.mod': 'go',
+            'go.sum': 'go'
+          };
+          
+          if (specialFiles[fileName]) {
+            fileExtensions[specialFiles[fileName]] = (fileExtensions[specialFiles[fileName]] || 0) + 1;
+          } else if (fileName.length > 0 && !fileName.startsWith('.')) {
+            // Likely an executable or script
+            fileExtensions['script'] = (fileExtensions['script'] || 0) + 1;
+          }
+        }
+      }
+    });
+    
+    // Map extensions to languages - expanded for CLI applications
+    const languageMap: { [key: string]: string } = {
+      // JavaScript/TypeScript
+      'js': 'JavaScript',
+      'mjs': 'JavaScript',
+      'cjs': 'JavaScript',
+      'jsx': 'JavaScript',
+      'ts': 'TypeScript',
+      'tsx': 'TypeScript',
+      
+      // Python
+      'py': 'Python',
+      'py3': 'Python',
+      'pyw': 'Python',
+      'pyx': 'Python',
+      'pyi': 'Python',
+      
+      // Shell scripting
+      'sh': 'Shell',
+      'bash': 'Bash',
+      'zsh': 'Shell', 
+      'fish': 'Shell',
+      'ksh': 'Shell',
+      'csh': 'Shell',
+      'tcsh': 'Shell',
+      'command': 'Shell',
+      
+      // C/C++
+      'c': 'C',
```

Comment:
I think it will be nicer to go with something like
```js
const languageGroups: Record<string, string[]> = {
  JavaScript: ['js', 'mjs', 'cjs', 'jsx'],
  TypeScript: ['ts', 'tsx'],
  Python: ['py', 'py3', 'pyw', 'pyx', 'pyi'],
  Shell: ['sh', 'zsh', 'fish', 'ksh', 'csh', 'tcsh', 'command'],
  Bash: ['bash'],
  C: ['c', 'h'],
  'C++': ['cpp', 'cxx', 'cc', 'c++', 'hpp', 'hxx', 'hh'],
  Rust: ['rs'],
  Go: ['go'],
  Zig: ['zig'],
  Nim: ['nim'],
} 
```
and then flatten it
```js
const languageMap: Record<string, string> = {};

for (const [language, extensions] of Object.entries(languageGroups)) {
  const cleanLanguage = language.replace(/Tooling$/, ''); // optional cleanup
  for (const ext of extensions) {
    languageMap[ext.toLowerCase()] = cleanLanguage;
  }
}
```

Reply:
This is more compact and still clear